CircleCI no_pic fails with:
```
  (HOST_DMD_RUN)  BACK_DOBJS  dmd/backend/evalu8.d
dmd -c -of../generated/linux/debug/32/evalu8.o  -version=MARS  -J../generated/linux/debug/32 -w -de -g -debug -g -m32 -mv=dmd.backend=dmd/backend -betterC dmd/backend/evalu8.d
dmd/root/array.d(19): Error: `TypeInfo` cannot be used with -betterC
posix.mak:552: recipe for target '../generated/linux/debug/32/evalu8.o' failed
make: *** [../generated/linux/debug/32/evalu8.o] Error 1
make: *** Waiting for unfinished jobs....
make: Leaving directory '/home/circleci/dmd/src'
Exited with code 2
```

I suppose that's because the backend is compiled with `-betterC`, while the frontend without. But I wonder why the issue manifests doesn't manifest itself on the CircleCI PIC - if anything it's a compile-time error, not link time.

--- a/src/dmd/backend/cgelem.d
+++ b/src/dmd/backend/cgelem.d
@@ -73,7 +73,7 @@ int REGSIZE();
 
 version (MARS)
 {
-extern void error(const(char)* filename, uint linnum, uint charnum, const(char)* format, ...);
+    import dmd.backend.errors;
 }
 
 /*****************************

--- a/src/dmd/backend/cgxmm.d
+++ b/src/dmd/backend/cgxmm.d
@@ -35,15 +35,14 @@ import dmd.backend.xmm;
 
 version (SCPP)
     import dmd.backend.exh;
-
+version (MARS)
+    import dmd.backend.errors;
 
 extern (C++):
 
 int REGSIZE();
 
 uint mask(uint m);
-
-void error(const(char)* filename, uint linnum, uint charnum, const(char)* format, ...);
 
 /*******************************************
  * Is operator a store operator?

--- a/src/dmd/backend/errors.di
+++ b/src/dmd/backend/errors.di
@@ -1,0 +1,17 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Copyright:   Copyright (C) 1999-2018 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/errors.d, _errors.d)
+ * Documentation:  https://dlang.org/phobos/dmd_errors.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/errors.d
+ */
+module dmd.backend.errors;
+
+/**
+   Print an error message, increasing the global error count
+ */
+extern (C++) void error(const(char)* filename, uint linnum, uint charnum, const(char)* format, ...);

--- a/src/dmd/backend/evalu8.d
+++ b/src/dmd/backend/evalu8.d
@@ -43,8 +43,8 @@ import scopeh;
 }
 
 extern (C++):
-
-extern void error(const(char)* filename, uint linnum, uint charnum, const(char)* format, ...);
+version (MARS)
+    import dmd.backend.errors;
 
 // fp.c
 int testFE();

--- a/src/dmd/backend/gother.d
+++ b/src/dmd/backend/gother.d
@@ -45,8 +45,8 @@ extern (C++):
 
 version (SCPP)
     import parser;
-
-extern void error(const(char)* filename, uint linnum, uint charnum, const(char)* format, ...);
+version (MARS)
+    import dmd.backend.errors;
 
 
 /**********************************************************************/

--- a/src/dmd/eh.d
+++ b/src/dmd/eh.d
@@ -18,6 +18,7 @@ import core.stdc.stdlib;
 import core.stdc.string;
 
 import dmd.globals;
+import dmd.errors;
 
 import dmd.backend.cc;
 import dmd.backend.cdef;
@@ -33,8 +34,6 @@ import dmd.backend.type;
 extern (C++):
 
 // Support for D exception handling
-
-void error(const(char)* filename, uint linnum, uint charnum, const(char)* format, ...);
 
 package(dmd) @property @nogc nothrow auto NPTRSIZE() { return _tysize[TYnptr]; }
 
@@ -378,4 +377,3 @@ void except_fillInEHTable(Symbol *s)
     assert(sz != 0);
     s.Sdt = dtb.finish();
 }
-

